src/SprykerEco/Zed/Inxmail/Business/Mapper/Order/AbstractOrderMapper.php

```diff
@@ -0,0 +1,308 @@
+                'Zip' => $orderTransfer->getShippingAddress()->getZipCode(),
+                'Country' => $orderTransfer->getShippingAddress()->getCountry()->getName(),
+            ],
+            'Order' => [
+                'Number' => $orderTransfer->getIdSalesOrder(),
+                'Comment' => $orderTransfer->getCartNote(),
+                'CancelComment' => '',
+                'OrderDate' => $this->dateTimeService->formatDateTime($orderTransfer->getCreatedAt()),
+                'SubTotal' => $this->getFormattedPriceFromInt($orderTransfer->getTotals()->getSubtotal()),
+                'Discount' => $this->getFormattedPriceFromInt($orderTransfer->getTotals()->getDiscountTotal()),
+                'Tax' => $this->getFormattedPriceFromInt($orderTransfer->getTotals()->getTaxTotal()->getAmount()),
+                'GrandTotal' => $this->getFormattedPriceFromInt($orderTransfer->getTotals()->getGrandTotal()),
+                'TotalDeliveryCosts' => $this->getDeliveryCosts($orderTransfer->getExpenses()),
+                'TotalPaymentCosts' => $this->getPaymentMethodsTotal($orderTransfer->getPayments()),
+            ],
+            'Payment' => $this->getPaymentMethodInfo($orderTransfer->getPayments()),
+            'Delivery' => $this->getOrderDeliveryInfo($orderTransfer->getShipmentMethods(), $orderTransfer->getExpenses()),
+            'Shop' => [
+                'ShopLocale' => $this->localeFacade->getCurrentLocaleName(),
+                'ShopUrl' => $this->config->getHostYves(),
+            ],
+        ];
+
+        foreach ($orderTransfer->getItems() as $item) {
+            $payload['OrderItem'][] = [
+                'Name' => $item->getName(),
+                'Sku' => $item->getSku(),
+                'Image' => $this->getItemImageLink($item->getImages()),
+                'DeepLink' => $this->getDeepLink($item, $locale),
+                'Price' => $this->getFormattedPriceFromInt($item->getRefundableAmount()),
+                'Quantity' => $item->getQuantity(),
+                'Sum' => $this->getFormattedPriceFromInt($item->getSumPriceToPayAggregation()),
+                'OriginalPrice' => $this->getFormattedPriceFromInt($item->getUnitPrice()),
+                'TaxAmount' => $this->getFormattedPriceFromInt($item->getUnitTaxAmount()),
+                'TaxRate' => $item->getTaxRate(),
+                'Discount' => $this->getFormattedPriceFromInt($item->getUnitDiscountAmountAggregation()),
+            ];
+        }
+
+        return $payload;
+    }
+
+    /**
+     * @return string
+     */
+    abstract protected function getEvent(): string;
+
+    /**
+     * @param \ArrayObject $images
+     *
+     * @return string
+     */
+    protected function getItemImageLink(ArrayObject $images): string
+    {
+        /**
+         * @var \Generated\Shared\Transfer\ProductImageTransfer $image
+         */
+        foreach ($images as $image) {
+            return $image->getExternalUrlSmall();
```

Comment:
What about case when you have a lot of images?

Reply:
I need only first image